src/realm.h

```diff
@@ -845,6 +850,28 @@ RLM_API bool realm_scheduler_set_notify_callback(realm_scheduler_t*, void* userd
  */
 RLM_API realm_t* realm_open(const realm_config_t* config);
 
+/**
+ * Deletes the following files for the given `realm_file_path` if they exist:
+ * - the Realm file itself
+ * - the .management folder
+ * - the .note file
+ * - the .log file
+ *
+ * The .lock file for this Realm cannot and will not be deleted as this is unsafe.
+ * If a different process / thread is accessing the Realm at the same time a corrupt state
+ * could be the result and checking for a single process state is not possible here.
+ *
+ * @param realm_file_path The path to the Realm file. All files will be derived from this.
+ * @param[out] did_delete_realm If non-null, set to true if the primary Realm file was deleted.
```

Comment:
Might be worth pointing out that this may be `true` even if an error does occur as it's somewhat counterintuitive.